R/mock-data.R

```diff
@@ -0,0 +1,134 @@
+#' Generate mock gff 
+#'
+#' Quick function to generate a mock gff. 
+#' 
+#' @param n_genes Number of genes in mock gff file
+#' @param chromosome Chromosome value. Default=1.
+#' 
+#' @return data.frame containing mock gff data from chromosome 1.
+#' 
+#' @export
+#' 
+mock_gff <- function(n_genes=1000, chromosome=1){
+    
+    mock_gff <- data.frame(list(V1 = chromosome,
+                                V2 = "source",
+                                V3 = "gene",
+                                V4 = sort(sample(1e4:2e8, n_genes))))
+    mock_gff$V5 <- mock_gff$V4 + floor(rnorm(n_genes, 1500, 1000))
+    mock_gff[, c("V6", "V7", "V8", "V9")] <- "."
+    
+    return(mock_gff)
+}
+
+
+#' Generate mock vcf 
+#'
+#' Quick function to generate mock vcf file. Note this data has unrealistic
+#' population structure.
+#' 
+#' @param n_snps Number of SNPs in mock vcf file. Default=10,000.
+#' @param n_samples Number of samples in mock bulk data. Default=10.
+#' @param chromosome Chromosome value. Default=1.
+#' 
+#' @return data.frame containing mock gff data from chromosome 1.
+#'
+#' @export
+#' 
+mock_vcf <- function(n_snps=1e4, n_samples=10, chromosome=1){
+    
+    if (requireNamespace("vcfR", quietly = TRUE))
+    data(vcfR_example)
```

Comment:
See comment below

Reply:
see comment below.

Reply:
I did change this to data(vcfR_example, envir = environment()) so that the example vcf data is loaded into the evaluation environment and won't take up space or replace existing objects in user's environments. Hoping that solves the issues!